File: app/src/main/java/de/christinecoenen/code/zapp/app/mediathek/repository/MediathekSearchSuggestionsProvider.java
package de.christinecoenen.code.zapp.app.mediathek.repository;

import android.content.Context;
import android.content.SearchRecentSuggestionsProvider;
import android.provider.SearchRecentSuggestions;


public class MediathekSearchSuggestionsProvider extends SearchRecentSuggestionsProvider {

	public final static String AUTHORITY = "de.christinecoenen.code.zapp.MediathekSearchSuggestionsProvider";
	public final static int MODE = DATABASE_MODE_QUERIES;

	public static void saveQuery(Context context, String query) {
		SearchRecentSuggestions suggestions = new SearchRecentSuggestions(context, AUTHORITY, MODE);
		suggestions.saveRecentQuery(query, null);
	}

	public MediathekSearchSuggestionsProvider() {
		setupSuggestions(AUTHORITY, MODE);
	}
}

File: app/src/main/java/de/christinecoenen/code/zapp/app/mediathek/repository/MediathekSearchSuggestionsProvider.kt
package de.christinecoenen.code.zapp.app.mediathek.repository

import android.content.Context
import android.content.SearchRecentSuggestionsProvider
import android.provider.SearchRecentSuggestions

class MediathekSearchSuggestionsProvider : SearchRecentSuggestionsProvider() {

	companion object {

		private const val AUTHORITY = "de.christinecoenen.code.zapp.MediathekSearchSuggestionsProvider"
		private const val MODE = DATABASE_MODE_QUERIES

		fun saveQuery(context: Context, query: String) {
			val suggestions = SearchRecentSuggestions(context, AUTHORITY, MODE)
			suggestions.saveRecentQuery(query, null)
		}

		@JvmStatic
		fun deleteAllQueries(context: Context) {
			val suggestions = SearchRecentSuggestions(context, AUTHORITY, MODE)
			suggestions.clearHistory()
		}
	}

	init {
		setupSuggestions(AUTHORITY, MODE)
	}
}
